api/pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -947,6 +981,20 @@ type ProtoPort struct {
 	Net string `json:"net,omitempty"`
 }
 
+// +kubebuilder:validation:Pattern=`^(([0-9]*(\.[0-9]*)?(ms|s|h|m|us)+)+|auto)$`
+type BPFConntrackTimeout string
+
+type BPFConntrackTimeouts struct {
+	CreationGracePeriod *BPFConntrackTimeout `json:"creationGracePeriod,omitempty"`
```

Comment:
Each of these fields should have a comment explaining what it does and what its default value is. That way our docs generation will build good human readable docs. I think as you've structured it now, this is all front-loaded on the `BPFConntrackTimeouts` structure instead of on the fields.